src/LanguageServer/Impl/Implementation/Server.cs

```diff
@@ -250,6 +250,7 @@ private sealed class ReloadModulesQueueItem : IAnalyzable {
             var reanalyze = true;
             if (@params.settings != null) {
                 if (@params.settings is ServerSettings settings) {
+                    Analyzer.EnableDiagnostics = settings.analysis.diagnostics;
```

Comment:
The setting needs to be added to the `package.json` in the extension defaulting to `true`

Reply:
Yep, I've done that locally for now (until I PR it), since I wasn't 100% sure on the naming.

Reply:
@DonJayamanne can suggest name of the setting